### ISSUE TITLE
RemoteStreams are getting disconnected abruptly and nullifies the stream in group call of 25+ participants

### ISSUE THREAD

**Issue by kamesh-a** [twilio/twilio-video.js#765]
- [x] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [x] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.
 - [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

Hello Twilio Team,

I would like to report an issue which is happening with 30 participants in call, Call starts properly and people are able join, after some point of time. All the participants are getting blacked out. Typically the mediaStream are disconnected due to video tags are either retained in last frame or pitch black.

We have also seen the WebSockets are getting disconnected directly **closed** event without reconnecting with heart beats and trying to renegotiation applying SDP.

Network is different for each person, some are in our office and some are working remotely from `uk`, `us` and home. so, this shouldn't possibly be a network effect.

This sequence happens only in Larger group calls 25 - 35.

#### Twilio Logs

[user-1](https://connect-feedback-logs.storage.googleapis.com/generallog-3151c946-19f2-4e02-98af-eedabae52abf.log)
[user-2](https://connect-feedback-logs.storage.googleapis.com/generallog-126045a9-eccf-4049-8d9f-ea08f52e7de0.log)
[user-3](https://connect-feedback-logs.storage.googleapis.com/generallog-37dbddff-b388-49fe-9390-7a8a01dc4088.log)
[user-4](https://connect-feedback-logs.storage.googleapis.com/generallog-872e1a61-e99b-4df8-b29b-cd55c7ec7344.log)
[user-5](https://connect-feedback-logs.storage.googleapis.com/generallog-b7633c17-0ce1-4af9-af6b-e34f120da6b6.log)

**Expected behavior:**
Group Call should not disconnect with Remote Participant for everyone in call at the same time
**Actual behavior:**
Group Call is disconnecting abruptly and video stream source receives null byes displaying black screen 

**Software versions:**

 - [x] Browser(s): Chrome, Firefox
 - [x] Operating System: Mac, Windows
 - [x] twilio-video.js: `2.0.0-beta12`
 - [x] Third-party libraries (e.g., Angular, React, etc.): React 16.8

**Comment by kamesh-a** [comment]
Hi @manjeshbhargav,
Can someone please take a look at this and update us.

Thanks.

**Comment by manjeshbhargav** [comment]
Hi @kamesh-a ,

Sorry for the delayed response. I checked the Room logs and this looks like all participants were using 2.0.0-beta12 instead of 2.0.0-beta14. I also noticed that while a lot of Participants experienced ICE failures, only a few managed to attempt reconnection.

I have to look at it more deeply, but on the surface this may be due to CPU / network congestion issues that may happen due to having to receive and render so many video tracks. Since you are already looking to use 2.0.0-beta14, I highly recommend that you try out the [Bandwidth Profile API](https://github.com/twilio/twilio-video.js/blob/2.0.0-beta14/CHANGELOG.md#200-beta14-september-17-2019), which will help you better sustain a Room with a large number of Participants.

Thanks,

Manjesh Malavalli
JSDK Team

**Comment by kamesh-a** [comment]
Hi @manjeshbhargav 

Thank you for looking into it, We really appreciate Twilio video team hardwork on making this Abstracted API work on cross browsers. Actually we are testing Bandwidth Profile API and we figure which layout is going to best suit our needs. 

Next week we are planning to release to our internal team and give feedback details how we implemented them to twilio team.

I will forward our usage feedbacks.

BTW,

`this may be due to CPU / network congestion issues` - I doubt this one, because that call was happening on high end CPU mostly Macbook pro and it happened in different network location at the same time, participants were in US (2), UK(2) and few members from remote work location in India ( Pune, Chennai ) from their homes all of them reported the same, when we checked remote stream tracks nullified abruptly.

Having said, we are definitely working on pushing Bandwidth Profile API to beta group, but it will take sometime as there two layout which we support in app ( Presentation Mode and Collaboration mode ). 

One caveat we have to select the mode of layout in Bandwidth Profile API while joining to room and not dynamically changeable, we are figuring out these parts.

Thanks
